---
title: "Fungal Beginnings"
author: "Scott Klasek"
date: '2022-08-12'
output: github_document
---
Purpose: fungal analysis for all 2018-2020 samples across the entire platform. Include Larkin's samples here, because they're all in potato. Lots of this is built off the previous document, 04_add_potato_data, but this one aims to be more a final, presentable product.  

#### load libraries
```{r}
library(tidyverse)
library(phyloseq)
library(patchwork)
library(vegan)
library(DESeq2)
library(metagenomeSeq)
library(NatParksPalettes)
library(UpSetR)
```

#### define functions
put functions before data import because some functions are used *for* data import
```{r}
# merge_with_jim_data takes as input a phyloseq object and a dataframe of objective 1 data that Jim has curated, which can be subsetted
# it writes data from Jim's spreadsheet into the corresponding phyloseq object, based on combinations
# make sure jim's data is cleaned up according to the steps above!
merge_with_jim_data <- function(ps){
  df <- left_join(data.frame(ps@sam_data), jim.info.s, 
                by = c("state" = "State", "objective" = "Objective", "rotation" = "Rotation",
                       "plot" = "Plot", "year" = "Year", "month" = "Month")) # merges 
  #the unique combo of columns from phyloseq sample data and jim's data
  rownames(df) <- rownames(ps@sam_data) # ok as long as the rows are in the same order, which they are
  sample_data(ps) <- df # put df into the sample_data
  return(ps)
}

# assign ASV numbers cuz i'm sick of the sequences as the rownames
number_asvs <- function(ps){
  dna <- Biostrings::DNAStringSet(taxa_names(ps))
  names(dna) <- taxa_names(ps)
  ps <- merge_phyloseq(ps, dna)
  taxa_names(ps) <- paste0("ASV", seq(ntaxa(ps)))
  return(ps)
}

# investigate library sizes
plot_libsize <- function(ps){
  libsizes <- as.data.frame(sample_data(ps)) # make data frame
  libsizes$LibrarySize <- sample_sums(ps) # obtain library size
  libsizes <- libsizes[order(libsizes$LibrarySize),] # order data frame
  libsizes$Index <- seq(nrow(libsizes)) # index samples for plotting
  gg <- ggplot(libsizes, aes(x=Index, y=LibrarySize)) + # make plot
    geom_point() + 
    geom_hline(yintercept=10000,linetype=2)
  return(gg)
}

# compare_taxa_abunds makes a dataframe of the percent abundances of a certain taxon at whatever taxonomic level
# and uses sample info to return data to test differences between groups of samples
compare_taxa_abunds <- function(ps, taxlevel, taxon){
  tax.df <- data.frame(ps@tax_table) # make taxonomy table a dataframe
  asvs <- rownames(tax.df[which(tax.df[taxlevel]==taxon),]) # get the ASVs that correspond to the taxon of interest
  percent.per.sample <- rowSums(ps@otu_table[,asvs]) / sample_sums(ps) * 100 # percentages per sample
  df <- cbind(data.frame(ps@sam_data) %>% select(state, year, season, block, in.potato, post.fumigation), percent.per.sample)
  return(df)
}

# hellinger transformation
helltransform <- function(ps){
  otu.ps3hell <- otu_table(decostand(otu_table(ps), method = "hellinger"), taxa_are_rows=FALSE)
  ps.hel <- phyloseq(tax_table(ps),
                    sample_data(ps),
                    otu_table(otu.ps3hell),
                    refseq(ps))
  return(ps.hel)
}

# asvcumulate calculates prevalence and cumulative abundance of ASVs, ranking by most common to least common, returning a dataframe
# input: a phyloseq object
asvcumulate <- function(ps){
  ps.cumul.abund <- data.frame(taxa_sums(ps)) # make a dataframe of the number of reads for each ASV across the dataset
  colnames(ps.cumul.abund)[1] <- "reads" # relabel the column
  ps.cumul.abund$pctreads <- 100*ps.cumul.abund$reads/sum(taxa_sums(ps)) # this column is the percent of reads in the entire dataset from each ASV
  asv.binary.ps <- data.frame(otu_table(ps)) # write ASV table to data frame
  asv.binary.ps[asv.binary.ps > 0] <- 1 # convert all reads to presence/absence
  asv.prevalence <- colSums(asv.binary.ps) # count the number of communities each ASV appears in 
  numsamples <- nrow(sample_data(ps)) # nsamples decided not to work so i had to add this in here (phyloseq v 1.34)
  asv.pct.prevalence <- (asv.prevalence/numsamples)*100 # convert this into percent
  ps.cumul.abund$pct_prevalence <- asv.pct.prevalence # add in prevalence information (what % of communities have this ASV in them)
  ps.cumul.abund <- ps.cumul.abund[order(-ps.cumul.abund$reads),] # this orders ASVs by most abundant to least (make sure you have more than one row in your dataframe when you do this)
  ps.cumul.abund$csum <- cumsum(ps.cumul.abund$pctreads) # calculate cumulative sum
  ps.cumul.abund$rank <- c(1:nrow(ps.cumul.abund)) # define a rank to plot with
  return(ps.cumul.abund)
}

# count the number of states in which an ASV was detected
num_states_per_asv <- function(ps){
  filter <- phyloseq::genefilter_sample(ps, filterfun_sample(function(x) x > 0)) # for each ASV get a true/false whether it has over 0 reads
  ps <- prune_taxa(filter, ps) # remove the zero-count taxa from the ps object 
  num_states <- vector("numeric") # define our numeric vector, and below: count the number of states in which an ASV is found in 
  for (i in rownames(ps@tax_table)){num_states[i] <- length(unique(ps@sam_data[rownames(ps@otu_table[which(ps@otu_table[,i] > 0),]),]$state))} 
  return(num_states)
}
```

#### import data  
1) Import Jim's yield and soil info data, do some clean up  
2) The fungal phyloseq object 
3) Add Jim's data to the phyloseq object using the custom function, merge_with_jim_data
At the end of this, we have only one input file, the phyloseq object, to work from.  
```{r}
### 
# Jim's yield and soil data to merge with phyloseq object metadata 
jim.info <- read.csv(file="/Users/klas0061/Desktop/UMN/jim_info/PSHP_ALL_obj_1_all_data_2022_07_08.csv") # import

# basic clean-up stuff
colnames(jim.info) <- jim.info[1,] # fix column names (there are two header cols)
jim.info <- jim.info[2:nrow(jim.info),] # omit redundant column name 
jim.info <- jim.info[,c(1:101,103:ncol(jim.info))] # omit the redundant year in column 102
for (i in 2:7){jim.info[,i] <- as.numeric(jim.info[,i])} # convert columns that should be numeric into numeric

# change OR Spring 2020 month to 4 so Cultivar (whether field is in potato or not) is not NA
jim.info[which(jim.info$State=="OR" & jim.info$Year==20 & jim.info$Month=="3"),"Month"] <- "4"

# select data columns to merge by, and to add (an example, can always select more later)
jim.info.s <- jim.info %>% select(State, Objective, Rotation, Plot, Year, Month, Cultivar, VPPG,
                                  `OM (%)`, `Total yield`, `Marketable yield`, `Greater than 6 oz.`) 

for (i in 8:12){jim.info.s[,i] <- as.numeric(jim.info.s[,i])} # again, convert columns that should be numeric into numeric

### import our phyloseq object of all fungi
all.fung.1920.ps  <- readRDS(file="/Users/klas0061/Desktop/UMN/phyloseqs/first_two_years_all_sites/all.fung1920.ps")

### add jim's data to the sample_data of phyloseq object
all.fung.1920.ps <- merge_with_jim_data(all.fung.1920.ps)
all.fung.1920.ps
```

#### more data cleaning
define column "in.potato", whether samples come from potato or rotation years
```{r}
# rewrite Cultivar column of sample data as in.potato
colnames(all.fung.1920.ps@sam_data)[which(colnames(all.fung.1920.ps@sam_data) == "Cultivar")] <- "in.potato"
# change the cultivar to "potato" because we have it in another column
all.fung.1920.ps@sam_data[which(all.fung.1920.ps@sam_data$in.potato != "Rotation"), "in.potato"] <- "Potato"
# Minnesota 2018 samples were NOT in potato
all.fung.1920.ps@sam_data[which(all.fung.1920.ps@sam_data$year==18), "in.potato"] <- "Rotation"
# larkin samples are ALL potato, except after harvest
all.fung.1920.ps@sam_data[which(all.fung.1920.ps@sam_data$state=="US"), "in.potato"] <- "Potato"
all.fung.1920.ps@sam_data[which(all.fung.1920.ps@sam_data$state=="US" & all.fung.1920.ps@sam_data$season=="Fall"), "in.potato"] <- "Rotation"
# make sure larkin treatment general_categories 3A and 3B are "amended", not "control"
all.fung.1920.ps@sam_data[which(all.fung.1920.ps@sam_data$state=="US" & all.fung.1920.ps@sam_data$treatment_new==3), "general_category"] <- "Amended"
```

For treatments that were fumigated, fumigation would have happened fall 2018 for the 3-year rotations, and fall 2019 for the two-year rotations. Add in another category "post.fumigation" that applies to in.potato samples from spring and summer that were fumigated the previous fall. 

```{r}
table(all.fung.1920.ps@sam_data %>% pull(general_category)) # breakdown of our general categories

# samples considered post.fumigation TRUE are 2019 3-yr rotation spring/summer, or 2020 2-yr spring/summer, and general_category = Fumigated or Amended/Fumigated
all.fung.1920.ps@sam_data$post.fumigation <- FALSE # write a new sample data column, post.fumigation is FALSE

all.fung.1920.ps@sam_data[which(all.fung.1920.ps@sam_data$rotation == 3 & all.fung.1920.ps@sam_data$year == 19 &
                                grepl("umigated", all.fung.1920.ps@sam_data$general_category) &
                                startsWith(all.fung.1920.ps@sam_data$season, "S")), "post.fumigation"] <- TRUE # post fumigation true for 3-yr rotations

all.fung.1920.ps@sam_data[which(all.fung.1920.ps@sam_data$rotation == 2 & all.fung.1920.ps@sam_data$year == 20 &
                                grepl("umigated", all.fung.1920.ps@sam_data$general_category) &
                                startsWith(all.fung.1920.ps@sam_data$season, "S")), "post.fumigation"] <- TRUE # post fumigation true for 2-yr rotations
```
*We now have several variables by which to examine community variation:*  
state, year, season, block, in.potato, and post.fumigation  
Keep in mind ALL post.fumigation samples are Potato, and that only Potato samples should be compared for fumigation effects.  

```{r}
summary(data.frame(all.fung.1920.ps@sam_data) %>% select(state, year, season, block, in.potato, post.fumigation)) # our variables summary

table(all.fung.1920.ps@sam_data$state) 
table(all.fung.1920.ps@sam_data$year) 
table(all.fung.1920.ps@sam_data$season)
table(all.fung.1920.ps@sam_data$block)
table(all.fung.1920.ps@sam_data$in.potato)
table(all.fung.1920.ps@sam_data$post.fumigation)
```
   
Different numbers of samples per state and block are roughly comparable and are probably trustworthy with permanovas. Season is ok too if comparing Spring vs Summer, which are in potato. (in.potato itself is confounded by season, Spring/Summer potato vs Fall rotation samples). For year, make sure to omit 2018 Minnesota samples. Be more careful with comparisons made between fumigated/unfumigated treatments, because sample sizes are definitely not balanced. 

#### Viridiplantae: to remove or not to remove? Some plots
```{r}
all.fung.1920.ps <- number_asvs(all.fung.1920.ps) # store the sequences as a biostrings object, and assign ASV numbers for ease

table(data.frame(all.fung.1920.ps@tax_table) %>% filter(Kingdom=="Viridiplantae") %>% pull(Phylum)) # the major ASVs are Anthophyta (real plants) but there are many ASVs belonging to Chlorophyta (green algae)

### plotting percent abundances of Viridiplantae by Phylum
# tax glom at phylum
all.fung.1920.phy <- tax_glom(all.fung.1920.ps, taxrank = "Phylum") # 

# transform to percent abundance
all.fung.1920.phy.pa <- transform_sample_counts(all.fung.1920.phy, function(OTU) OTU/sum(OTU) * 100)

# subset only the kingdom Viridiplantae
all.fung.1920.phy.pa <- subset_taxa(all.fung.1920.phy.pa, Kingdom == "Viridiplantae")

# plot percent abundances of phyla from kingdom Viridiplantae
plot_bar(all.fung.1920.phy.pa, fill="Phylum")+
   geom_bar(stat="identity", position="stack")+ 
   scale_fill_manual("Phylum", values = natparks.pals("Acadia", length(unique(all.fung.1920.phy.pa@tax_table[,"Phylum"]))))+
   facet_grid(~state, scales = "free", space = "free")+
   scale_y_continuous(limits = c(0,70))+
   ggtitle("")+
   theme(axis.text.x = element_blank())

### plotting percent abundances of different taxonomic levels across samples

# first the major phyla of Viridiplantae
gg.anth <- ggplot(compare_taxa_abunds(all.fung.1920.ps, "Phylum", "Anthophyta"), aes(state, percent.per.sample))+
  geom_jitter()+ggtitle("Anthophyta percent abundances")

gg.chlo <- ggplot(compare_taxa_abunds(all.fung.1920.ps, "Phylum", "Chlorophyta"), aes(state, percent.per.sample))+
  geom_jitter()+ggtitle("Chlorophyta percent abundances")

gg.anth + gg.chlo

gg.solan <- ggplot(compare_taxa_abunds(all.fung.1920.ps, "Genus", "Solanum"), aes(state, percent.per.sample, color=in.potato))+
  geom_jitter()+ggtitle("Solanum percent abundances")

gg.brass <- ggplot(compare_taxa_abunds(all.fung.1920.ps, "Genus", "Brassica"), aes(state, percent.per.sample, color=in.potato))+
  geom_jitter()+ggtitle("Brassica percent abundances")

gg.trit <- ggplot(compare_taxa_abunds(all.fung.1920.ps, "Genus", "Triticum"), aes(state, percent.per.sample, color=in.potato))+
  geom_jitter()+ggtitle("Triticum percent abundances")

gg.solan + gg.brass + gg.trit + plot_layout(guides = "collect")
```
Most of the Viridiplantae actually are Chlorophyta, green algae. Is this biologically relevant?  
The most abundant orders are: Solanales (no surprise), Caryophyllales, Poales, and Brassicales. 
Solanales: pretty much all potatoes as far as I can tell (mainly tuberosum or NA, some bukasovii)  
Caryophyllales: Amaranths, like Bassia scoparia (a weed), Chenopodium quinoa/acuminatum (quinoa and a wild relative?), Amaranthus deflexus (an amaranth), Salsola collina (saltwort), Fagopyrum (buckwheat), Portulaca oleraca (purslane)... so many of them may be weeds
Poales: Triticum (wheat), Hordeum (barley)
Brassicales: Raphanus (radish), Brassica carinata/napus/juncea (rape/mustard). 
So basically they represent target crops (potatoes), weeds, and rotation crops.  

From the plots above you can guess that Colorado is rotating with Brassica, and Michigan with wheat. 

#### Chlorophyta: big orders are Chlymadomonadales, Sphaeropleales
```{r}
gg.chlam <- ggplot(compare_taxa_abunds(all.fung.1920.ps, "Order", "Chlamydomonadales"), aes(state, percent.per.sample))+
  geom_jitter()+ggtitle("Chlamydomonadales percent abundances")
gg.sphae <- ggplot(compare_taxa_abunds(all.fung.1920.ps, "Order", "Sphaeropleales"), aes(state, percent.per.sample))+
  geom_jitter()+ggtitle("Sphaeropleales percent abundances")
gg.chlam + gg.sphae
```
#### Library size (before pruning any plant ITS sequences)
```{r}
plot_libsize(all.fung.1920.ps)+ggtitle("ITS library sizes, 2018-2020")
```
#### Cleanup
Remove NA samples, prune at some threshold, omit Phylum Anthophyta
```{r}
# remove OR blanks
all.fung.1920.c.ps <- subset_samples(all.fung.1920.ps, sample_names(all.fung.1920.ps) != 
                                       c("OR_1_2_blank_20_4_ITS_S1039", "OR_1_2_blank_20_6_ITS_S1040", "OR_1_3_blank_20_8_ITS_S1065")) 
# omit sequences belonging to true plants 
all.fung.1920.c.ps <- subset_taxa(all.fung.1920.c.ps, (Phylum!="Anthophyta")) 

plot_libsize(all.fung.1920.c.ps)+ggtitle("ITS library sizes from cleaned samples, 2018-2020") # looks nearly identical as previous

# prune samples below 10k reads
all.fung.1920.c.ps <- subset_samples(all.fung.1920.c.ps, sample_sums(all.fung.1920.c.ps) > 10000) # throws away 118 samples yikes

# write out phyloseq object
#saveRDS(all.fung.1920.c.ps, "/Users/scottklasek/Desktop/UMN/phyloseqs/fung1920.cleaned.ps") # cleaned fungal phyloseq object

# check the balance of sample numbers
table(all.fung.1920.c.ps@sam_data$state) / table(all.fung.1920.ps@sam_data$state) # proportions of libraries > 10k reads, by state
table(all.fung.1920.c.ps@sam_data$year) / table(all.fung.1920.ps@sam_data$year)
table(all.fung.1920.c.ps@sam_data$season) / table(all.fung.1920.ps@sam_data$season)
table(all.fung.1920.c.ps@sam_data$block) / table(all.fung.1920.ps@sam_data$block)
table(all.fung.1920.c.ps@sam_data$in.potato) / table(all.fung.1920.ps@sam_data$in.potato)
table(all.fung.1920.c.ps@sam_data$post.fumigation) / table(all.fung.1920.ps@sam_data$post.fumigation)
```

Doesn't seem to be huge bias in the libraries removed due to low reads, except mayyyybe by state?  

### ANALYSIS TIME

#### ITS alpha diversity
```{r}
# calculate the alpha diversity (three metrics)
fung.div <- estimate_richness(all.fung.1920.c.ps, measures = c("Observed", "Shannon", "InvSimpson")) 

# merge select sample data columns with the alpha diversity data, and clean up the dataframe
d <- merge(fung.div, data.frame(all.fung.1920.c.ps@sam_data)[,c("state", "year", "month", "season", "block", "cultivar", "in.potato", "OM....", "post.fumigation")], by=0)
rownames(d) <- d$Row.names
d <- d %>% select(-Row.names)

# plot the three metrics I've calculated
cols.threediscrete <- c("#1b9e77", "#d95f02", "#7570b3")
f1920obs <- ggplot(d, aes(state, Observed, color=as.character(year)))+geom_jitter(width=0.3, size = 0.9)+
  scale_x_discrete("", limits = c("OR", "ID", "CO", "MN", "ND", "WI", "MI", "ME", "US"), labels = c("OR", "ID", "CO", "MN1", "MN2", "WI", "MI", "ME1", "ME2"))+
  scale_y_continuous("")+
  scale_color_manual("Year", values = cols.threediscrete)+
  theme_bw()#+ggtitle("", subtitle = "B")

f1920sha <- ggplot(d, aes(state, Shannon, color=as.character(year)))+geom_jitter(width=0.3, size = 0.9)+
  scale_x_discrete("", limits = c("OR", "ID", "CO", "MN", "ND", "WI", "MI", "ME", "US"), labels = c("OR", "ID", "CO", "MN1", "MN2", "WI", "MI", "ME1", "ME2"))+
  scale_y_continuous("")+
  scale_color_manual("Year", values = cols.threediscrete)+
  theme_bw()+theme(legend.position = "none")#+ggtitle("", subtitle = "D")

f1920sim <- ggplot(d, aes(state, InvSimpson, color=as.character(year)))+geom_jitter(width=0.3, size = 0.9)+
  scale_x_discrete("", limits = c("OR", "ID", "CO", "MN", "ND", "WI", "MI", "ME", "US"), labels = c("OR", "ID", "CO", "MN1", "MN2", "WI", "MI", "ME1", "ME2"))+
  scale_y_continuous("")+
  scale_color_manual("Year", values = cols.threediscrete)+
  theme_bw()+theme(legend.position = "none")#+ggtitle("", subtitle = "F")

f1920obs / f1920sha / f1920sim + plot_layout(guides = "collect")
```
Now for some follow-up stats on the alpha diversity
```{r}
summary(aov(Observed~state, data=d)) # state is significant across all
summary(aov(Shannon~state, data=d))
summary(aov(InvSimpson~state, data=d))

summary(aov(Observed~year, data=d)) # year is significant across all
summary(aov(Shannon~year, data=d))
summary(aov(InvSimpson~year, data=d))

# for all POTATO soils in states that fumigated, are there differences across fumigations?
unique(d %>% filter(post.fumigation==TRUE) %>% pull(state)) # all states except CO and US fumigated
alpha.fum.d <- d %>% filter(state != "US" & state != "CO" & in.potato == "Potato") # write them in a separate dataframe

# apparently WI fumigated the entire field before planting in 2020, change this:
alpha.fum.d[which(alpha.fum.d$state=="WI" & alpha.fum.d$year==20),"post.fumigation"] <- TRUE
alpha.fum.d[which(alpha.fum.d$state=="WI" & alpha.fum.d$year==19),"post.fumigation"] <- FALSE

# and ND did the reverse
alpha.fum.d[which(alpha.fum.d$state=="ND" & alpha.fum.d$year==19),"post.fumigation"] <- TRUE
alpha.fum.d[which(alpha.fum.d$state=="ND" & alpha.fum.d$year==20),"post.fumigation"] <- FALSE

table(alpha.fum.d$post.fumigation) # remember nearly twice as many communities are unfumigated

t.test(alpha.fum.d %>% filter(post.fumigation == TRUE) %>% pull(Observed),
       alpha.fum.d %>% filter(post.fumigation == FALSE) %>% pull(Observed))

t.test(alpha.fum.d %>% filter(post.fumigation == TRUE) %>% pull(Shannon),
       alpha.fum.d %>% filter(post.fumigation == FALSE) %>% pull(Shannon))

t.test(alpha.fum.d %>% filter(post.fumigation == TRUE) %>% pull(InvSimpson),
       alpha.fum.d %>% filter(post.fumigation == FALSE) %>% pull(InvSimpson))

gg.fum.alpha <- ggplot(alpha.fum.d %>% pivot_longer(cols = c("Observed", "Shannon", "InvSimpson"), names_to = "diversity"), 
       aes(state, value, color=post.fumigation))+
  geom_jitter(width=0.3, size = 0.9)+
  scale_x_discrete("State", limits = c("OR", "ID", "ND", "MN", "WI", "MI", "ME"))+
  scale_y_continuous("Diversity metric")+
  scale_color_discrete("Fumigation")+
  facet_grid(diversity~factor(year, levels = c(19,20), labels = c("2019", "2020")), scales = "free")+
  theme_bw()+ggtitle("Eukaryotic alpha diversity")
gg.fum.alpha

# Michigan
# t.test(alpha.fum.d %>% filter(state=="MI" & year==19) %>% pull(InvSimpson), # generally little difference in alpha diversity by year
#        alpha.fum.d %>% filter(state=="MI" & year==20) %>% pull(InvSimpson))

t.test(alpha.fum.d %>% filter(state=="MI" & post.fumigation == FALSE) %>% pull(Observed),
       alpha.fum.d %>% filter(state=="MI" & post.fumigation == TRUE) %>% pull(Observed))

t.test(alpha.fum.d %>% filter(state=="MI" & post.fumigation == FALSE) %>% pull(Shannon),
       alpha.fum.d %>% filter(state=="MI" & post.fumigation == TRUE) %>% pull(Shannon))

t.test(alpha.fum.d %>% filter(state=="MI" & post.fumigation == FALSE) %>% pull(InvSimpson),
       alpha.fum.d %>% filter(state=="MI" & post.fumigation == TRUE) %>% pull(InvSimpson))

# Minnesota
# t.test(alpha.fum.d %>% filter(state=="MN" & year==19) %>% pull(Shannon), # alpha diversity is different by year
#        alpha.fum.d %>% filter(state=="MN" & year==20) %>% pull(Shannon))

t.test(alpha.fum.d %>% filter(state=="MN" & post.fumigation == FALSE) %>% pull(Observed),
       alpha.fum.d %>% filter(state=="MN" & post.fumigation == TRUE) %>% pull(Observed))

t.test(alpha.fum.d %>% filter(state=="MN" & post.fumigation == FALSE) %>% pull(Shannon),
       alpha.fum.d %>% filter(state=="MN" & post.fumigation == TRUE) %>% pull(Shannon))

t.test(alpha.fum.d %>% filter(state=="MN" & post.fumigation == FALSE) %>% pull(InvSimpson),
       alpha.fum.d %>% filter(state=="MN" & post.fumigation == TRUE) %>% pull(InvSimpson))

# ND
t.test(alpha.fum.d %>% filter(state=="ND" & post.fumigation == FALSE) %>% pull(Observed),
       alpha.fum.d %>% filter(state=="ND" & post.fumigation == TRUE) %>% pull(Observed))

t.test(alpha.fum.d %>% filter(state=="ND" & post.fumigation == FALSE) %>% pull(Shannon),
       alpha.fum.d %>% filter(state=="ND" & post.fumigation == TRUE) %>% pull(Shannon))

t.test(alpha.fum.d %>% filter(state=="ND" & post.fumigation == FALSE) %>% pull(InvSimpson),
       alpha.fum.d %>% filter(state=="ND" & post.fumigation == TRUE) %>% pull(InvSimpson))
```
Alpha diversity varies across states and years, and does not across block (fortunately). Whether or not a plot was in potato was less clear, with different metrics giving different results. In states that had fumigation treatments and were in potato, t-tests say that fumigated communities unambiguously (ie all three metrics agreed) had lower measurements of alpha diversity, but wait a second-- when plotting by state and year, it seems like fumigation only reduces alpha diversity in Minnesota and Michigan.  

In Michigan, alpha diversity was largely not different across years, so I t-tested all samples together and for each metric, fumigated samples had lower diversity.  
In Minnesota, diversity was higher in 2019, so I tested each year separately. For each metric, fumigated samples had lower diversity.  

#### ordinations
```{r}
# transform
all.fung.1920.c.ps.hel <- helltransform(all.fung.1920.c.ps) # hellinger transformation

# make distance matrices
dm.fung1920.bjac <- phyloseq::distance(all.fung.1920.c.ps, method = "jaccard", binary = TRUE) # binary jaccard
dm.fung1920.hel.bc <- phyloseq::distance(all.fung.1920.c.ps.hel, method = "bray") # bray-curtis with hellinger

# ordinate
ord.fung1920.hel.bc.pcoa <- ordinate(all.fung.1920.c.ps.hel, method="PCoA", distance=dm.fung1920.hel.bc) # hellinger
ord.fung1920.bjac.pcoa <- ordinate(all.fung.1920.c.ps, method="PCoA", distance=dm.fung1920.bjac) # binary jaccard

# plots
cols.alphabetical <- c("#236192", "#B3A369", "#B0D7FF", "#18453B", "#7A0019", "#FFC72A", "#DC4405", "gray", "#C5050C")
gg.bc.ord <- plot_ordination(all.fung.1920.c.ps.hel, ord.fung1920.hel.bc.pcoa, color="state", title="ITS PCoA, Bray-Curtis, Hellinger transformation")+
  scale_color_manual(values = cols.alphabetical)
gg.bjac.ord <- plot_ordination(all.fung.1920.c.ps, ord.fung1920.bjac.pcoa, color="state", title="ITS PCoA, Binary Jaccard")+
  scale_color_manual(values = cols.alphabetical)
gg.bc.ord + gg.bjac.ord + plot_layout(guides = "collect")
```
Similar patterns seen as before, and the Larkin USDA samples do plot closely with U of Maine samples as I suspected. Other variables don't show much, if any, separation.  
Now let's see which variables explain variance among communities (this chunk takes a while):  
```{r}
# first by state
adonis.state <- adonis2(formula = dm.fung1920.hel.bc ~ state, data = data.frame(all.fung.1920.c.ps@sam_data)) # 59.7% of variance is by state!
adonis.state

# drop certain samples for certain comparisons

# blocks, because Colorado
subset.ps <- subset_samples(all.fung.1920.c.ps, state !="CO") # removing CO because it has no blocks
subset.ps <- helltransform(subset.ps) # hellinger transformation
dm.subset <- phyloseq::distance(subset.ps, method = "bray") # bray-curtis distance matrix
adonis.block <- adonis2(formula = dm.subset ~ block, data = data.frame(subset.ps@sam_data)) # compare blocks
adonis.block # block is 0.4% of variance, that's good

# year, because Minnesota 2018
subset.ps <- subset_samples(all.fung.1920.c.ps, year !=18) # removing MN 2018 samples
subset.ps <- helltransform(subset.ps) # hellinger transformation
dm.subset <- phyloseq::distance(subset.ps, method = "bray") # bray-curtis distance matrix
adonis.year <- adonis2(formula = dm.subset ~ year, data = data.frame(subset.ps@sam_data)) # compare years
adonis.year # 0.9% of variance by year

# season, in.potato
subset.ps <- subset_samples(all.fung.1920.c.ps, in.potato == "Potato") # removing fall samples that are not in potato
subset.ps <- helltransform(subset.ps) # hellinger transformation
dm.subset <- phyloseq::distance(subset.ps, method = "bray") # bray-curtis distance matrix
adonis.season <- adonis2(formula = dm.subset ~ season, data = data.frame(subset.ps@sam_data)) # compare spring/summer seasons
adonis.season # 0.6% of variance between spring/summer

# post.fumigation, for in.potato (unbalanced!)
# fix the WI fumigation fiasco:
subset.ps@sam_data[which(subset.ps@sam_data$state=="WI" & subset.ps@sam_data$year==20),"post.fumigation"] <- TRUE
subset.ps@sam_data[which(subset.ps@sam_data$state=="WI" & subset.ps@sam_data$year==19),"post.fumigation"] <- FALSE
adonis.fum <- adonis2(formula = dm.subset ~ post.fumigation, data = data.frame(subset.ps@sam_data)) 
adonis.fum # 3.8% of variance

# potato in spring/summer vs rotation in fall (unbalanced!)
adonis.potato <- adonis2(formula = dm.fung1920.hel.bc ~ in.potato, data = data.frame(all.fung.1920.c.ps@sam_data))
adonis.potato # 0.6% of variance 
```
In conclusion: state is by far the largest contributor of variance, then it seems as though fumigation matters a little. Then year at just under 1%, and block the least at 0.4%. There's probably a better way to do this somewhere, but the main point is clear for now.  

#### occupancy
```{r}
asvc.fungi <- asvcumulate(all.fung.1920.c.ps) # get prevalence and mean abundances of each fungal ASV
summary(asvc.fungi %>% pull(pct_prevalence)) # nobody is core, most are in only a few samples

# count the number of states in which an ASV was detected
num_states_per_asv_test <- num_states_per_asv(all.fung.1920.c.ps) # took like 8 minutes

# put the num_state count into the asvc dataframe
test_merge <- merge(asvc.fungi, data.frame(num_states_per_asv_test), by=0) 

# plot abundance occupancy
aostate <- ggplot(test_merge, aes(log10(pctreads), pct_prevalence, color=num_states_per_asv_test))+
  geom_point()+
  scale_color_continuous("Number of states detected")+
  scale_x_continuous("log mean abundance")+
  scale_y_continuous("percent occupancy")+
  ggtitle("Fungal occupancy / abundance")+theme_bw()
aostate
```

```{r}
hist(test_merge %>% filter(pctreads > 0.1) %>% pull(pct_prevalence), breaks=25,
     main = "Occupancy of high-abundance ASVs", xlab = "Percent occupancy")
```
Clear bimodal pattern: The ASVs on the left are really high-abundance, but only in samples from a few states, while the ones on the right are in pretty high abundance across the continent. 60% occupancy seems like a natural cutoff.   

```{r}
ggplot(test_merge, aes(num_states_per_asv_test, log10(pctreads)))+
  geom_jitter()+
  scale_y_continuous("log10 percent abundance")+
  scale_x_continuous("Number of states ASV found in", breaks = c(1:9))+
  theme_bw()
```


#### bar plots of cosmopolitan and state/regionally specific taxa  
```{r}
# remove nonexistent ASVs that must have been from the blank samples
filter <- phyloseq::genefilter_sample(all.fung.1920.c.ps, filterfun_sample(function(x) x > 0)) # for each ASV get a true/false whether it has over 0 reads
all.fung.1920.c.ps <- prune_taxa(filter, all.fung.1920.c.ps) # remove the zero-count taxa from the ps object

# transform to percent abundance
fung1920.pa <- transform_sample_counts(all.fung.1920.c.ps, function(OTU) OTU/sum(OTU) * 100)

# try putting the asvc (as a matrix) into the tax table of the resulting object and graphing by "taxa" as number of states each ASV was found in
rownames(test_merge) <- test_merge$Row.names
state.count <- test_merge %>% select(num_states_per_asv_test) # does this need to be in the same order as taxa are in ps?
state.count$num_states_per_asv_test <- as.character(state.count$num_states_per_asv_test) # needs to be a character matrix for phyloseq tax table

# warning: stupid bullshit
# there's a glitch in phyloseq where you can't plot or subset by the leftmost column of a tax table, so I need to throw in another bogus one.
dummy <- rep("Eukarya", times=nrow(fung1920.pa@tax_table))
dummy <- as.matrix(dummy)
rownames(dummy) <- rownames(fung1920.pa@tax_table)
tmerge <- merge(dummy, as.matrix(state.count), by=0)
rownames(tmerge) <- tmerge$Row.names
tmerge <- tmerge %>% select(V1, num_states_per_asv_test)
# stupid bullshit over

# add in the stupid dataframe containing state-counts as our tax table
fung1920.pa.statecount <- phyloseq(tax_table(as.matrix(tmerge)),
                                   otu_table(fung1920.pa),
                                   sample_data(fung1920.pa))

# so we tax-glom before plotting to make it easier on our hardworking computer
fung1920.pa.statecount <- tax_glom(fung1920.pa.statecount, taxrank = "num_states_per_asv_test")

gg.bar.statecount <- plot_bar(fung1920.pa.statecount, fill="num_states_per_asv_test")+
   geom_bar(stat="identity", position="stack")+
   scale_fill_manual("ASVs present in\n this many states", values = natparks.pals("Acadia", 9))+
   facet_grid(~factor(state, levels = c("OR", "ID", "CO", "MN", "ND", "WI", "MI", "ME", "US"), labels = c("OR", "ID", "CO", "MN1", "MN2", "WI", "MI", "ME1", "ME2")), scales = "free", space = "free")+
   scale_y_continuous(limits = c(0,100))+
   ggtitle("")+
   theme(axis.text.x = element_blank())
gg.bar.statecount

# what is causing the variability in some states? (ME, ND, and WI in particular)
fung1920.pa.statecount.wi.20 <- subset_samples(fung1920.pa.statecount, state=="WI" & year == 20)
wi20state <- plot_bar(fung1920.pa.statecount.wi.20, fill="num_states_per_asv_test")+
   geom_bar(stat="identity", position="stack")+
   scale_fill_manual("ASVs present in\n this many states", values = natparks.pals("Acadia", 9))+
   facet_grid(~factor(season, levels = c("Spring", "Summer", "Fall")), scales = "free", space = "free")+
   scale_y_continuous(limits = c(0,100))+
   ggtitle("WI 2020 samples")+
   theme(axis.text.x = element_blank())
fung1920.pa.statecount.wi.19 <- subset_samples(fung1920.pa.statecount, state=="WI" & year == 19)
wi19state <- plot_bar(fung1920.pa.statecount.wi.19, fill="num_states_per_asv_test")+
   geom_bar(stat="identity", position="stack")+
   scale_fill_manual("ASVs present in\n this many states", values = natparks.pals("Acadia", 9))+
   facet_grid(~factor(season, levels = c("Spring", "Summer", "Fall")), scales = "free", space = "free")+
   scale_y_continuous(limits = c(0,100))+
   ggtitle("WI 2019 samples")+
   theme(axis.text.x = element_blank())
wi19state + wi20state + plot_layout(guides = "collect")

fung1920.pa.statecount.mi <- subset_samples(fung1920.pa.statecount, state=="MI" & year==20)
mi.state <- plot_bar(fung1920.pa.statecount.mi, fill="num_states_per_asv_test")+
   geom_bar(stat="identity", position="stack")+
   scale_fill_manual("ASVs present in\n this many states", values = natparks.pals("Acadia", 9))+
   facet_grid(~post.fumigation, scales = "free", space = "free")+
   scale_y_continuous(limits = c(0,100))+
   ggtitle("MI 2020 samples by fumigation")+
   theme(axis.text.x = element_blank())
mi.state
```

#### rarefaction curves for WI diversity 
```{r eval=FALSE}
wi.rarefy.df <- rarecurve(otu_table(subset_samples(all.fung.1920.c.ps, state=="WI")), step=50, cex=0.5, tidy = TRUE) # make a dataframe of rarefaction counts for WI samples
wi.rarefy.df$year <- as.numeric(substr(wi.rarefy.df$Site, 12, 13)) # assign the year
wi.rarefy.df[which(is.na(wi.rarefy.df$year)),"year"] <- 20
wi.rarefy.df$month <- as.numeric(gsub("_.*$","",sub("..............", "", wi.rarefy.df$Site))) # assign the month
table(wi.rarefy.df$month)
unique(wi.rarefy.df %>% filter(month==2) %>% pull(Site))
wi.rarefy.df[which(wi.rarefy.df$month==2),"month"] <- 11 # fix mislabeled month
wi.rarefy.df$postfum <- ifelse(wi.rarefy.df$year==19, "unfumigated",
                               ifelse(wi.rarefy.df$year==20 & wi.rarefy.df$month < 11, "fumigated", "unfumigated")) # fumigated category

# plot
wi.raref.19 <- ggplot(wi.rarefy.df %>% filter(year==19), aes(Sample, Species, color=as.factor(month)))+
  geom_line(size=0.07)+
  scale_color_discrete("month")+
  scale_y_continuous("ASV richness")+
  scale_x_continuous("read number")+
  ggtitle("WI 2019 ITS")+
  geom_vline(xintercept = 10000, linetype="dotted")+
  theme_bw()

wi.raref.20 <- ggplot(wi.rarefy.df %>% filter(year==20), aes(Sample, Species, color=as.factor(month)))+
  geom_line(size=0.07)+
  scale_color_discrete("month")+
  scale_y_continuous("ASV richness")+
  scale_x_continuous("read number")+
  ggtitle("WI 2020 ITS")+
  geom_vline(xintercept = 10000, linetype="dotted")+
  theme_bw()

wi.raref.19 + wi.raref.20  
```
  This makes it seem like post-fumigation really did a number on the diversity, but remember this is CONFOUNDED with field effects. The two-year field could just have much lower diversity for other reasons... but this is still very low diversity and I'm tempted to think fumigation had to have made a difference.   

#### Make an upset plot 
```{r}
all.states.asvs <- rownames(test_merge %>% arrange(rank) %>% filter(num_states_per_asv_test == 9)) # sequences present in each state

# merge samples by state
ps.state.upset <- merge_samples(all.fung.1920.c.ps, sample_data(all.fung.1920.c.ps)$state, fun = sum)

# transposes the otu table and writes it as a dataframe
venn_obj <- as.data.frame(t(otu_table(ps.state.upset)))
venn_obj.binary <- sapply(venn_obj, function(x) ifelse(x > 0, 1, 0),
                          USE.NAMES = T) # makes the dataframe binary 
rownames(venn_obj.binary) <- rownames(venn_obj) # assigns the rownames as the same
venn_obj.binary <- as.data.frame(venn_obj.binary) # creates a dataframe where rows are ASVs, columns are block_env categories, and entries are either 1 or 0
upset_order <- colnames(venn_obj.binary) # assigns the order of sets in the upset plot as the column names

# plot
state.upset <- upset(venn_obj.binary, 
      sets = rev(upset_order),
      mainbar.y.label = 'ASV count',
      sets.x.label = 'Observed ASVs by site',
      order.by = 'freq', nintersects = 30, # can change order.by to "degree", or adjust number of intersections to show 
      keep.order = T)
state.upset # 
```
#### Who are the cosmopolitan fungi/eukaryotes?
```{r}
all.state.asvs.fung.1920.pa <- prune_taxa(all.states.asvs, fung1920.pa) # ps object with only the 9-state ASVs
length(unique(tax_table(all.state.asvs.fung.1920.pa)[,"Kingdom"]))

# plot by kingdom
plot_bar(all.state.asvs.fung.1920.pa, fill="Kingdom")+
   geom_bar(stat="identity", position="stack")+
   scale_fill_manual("ASVs present in\n this many states", values = natparks.pals("Yellowstone", 4))+
   facet_grid(~state, scales = "free", space = "free")+
   scale_y_continuous(limits = c(0,100))+
   ggtitle("Cosmopolitan Eukaryotic ASVs")+
   theme(axis.text.x = element_blank())

not.fungi <- rownames(data.frame(tax_table(all.state.asvs.fung.1920.pa)) %>% filter(Kingdom !="Fungi"))
all.state.asvs.fung.1920.pa@tax_table[not.fungi,]

# only fungi
all.state.asvs.actuallyfungi.1920.pa <- prune_taxa(setdiff(all.states.asvs, not.fungi), fung1920.pa) # ps
plot_bar(all.state.asvs.actuallyfungi.1920.pa , fill="Phylum")+
   geom_bar(stat="identity", position="stack")+
   scale_fill_manual("Phylum", values = natparks.pals("GrandCanyon", 6))+
   facet_grid(~state, scales = "free", space = "free")+
   scale_y_continuous(limits = c(0,80))+
   ggtitle("Cosmopolitan Fungal ASVs")+
   theme(axis.text.x = element_blank())

# by Class
plot_bar(all.state.asvs.actuallyfungi.1920.pa , fill="Class")+
   geom_bar(stat="identity", position="stack")+
   #scale_fill_manual("Phylum", values = natparks.pals("GrandCanyon", 6))+
   facet_grid(~state, scales = "free", space = "free")+
   scale_y_continuous(limits = c(0,80))+
   ggtitle("Cosmopolitan Fungal ASVs")+
   theme(axis.text.x = element_blank())

all.state.asvs.fung.1920.pa@tax_table[rownames(test_merge[all.states.asvs,])[1:7],] # taxonomy of the most abundant cosmopolitan fungi
```
   
Cosmopolitan eukaryotes consist largely of fungi, but there is one Nematode, two members of Ciliophora (protists) and that green algae Chlorosarcinopsis eremi.  

When looking at Fungal ASVs only, the most dominant phylum is Ascomycota. Western states except OR also have lots of Basidiomycota. Maine, USDA, and North Dakota also have some Mortierellomycota.  

It seems like little is known about the first two, P. pannorum is closely related to the fungus that causes white-nose disease in bats. M. elongata is apparently PGP for many diverse plants, [more info here](https://edis.ifas.ufl.edu/publication/SS679). It is highly abundant in soils,  associates with plant roots, and harbors bacteria inside its hyphae. The bacteria can modulate host metabolism... and a PGP mechanism is currently unknown. So this would be a very good candidate to examine cross-domain interactions: Is M. elongata associated with tuber yield? If so, which bacteria associate with it, or if only in certain states, which bacteria might be associated with it that could drive a growth-association?
 
#### another attempt at plotting percent abundances of core eukaryotic ASVs at Phylum level 
```{r}
fung1920.pa # percent abundance ps object with all taxa
ITS.ASVs.asvc <- read.csv(file = "/Users/klas0061/Desktop/UMN/figures/from_server/ITS.ASVs.asvc") # info with state-counts
core.its.asvs <- ITS.ASVs.asvc %>% filter(num_states_per_asv_test == 9) %>% pull(Row.names) # finally the core its asvs
all.state.asvs.fung.1920.pa <- prune_taxa(core.its.asvs, fung1920.pa) # ps object with only the 9-state ASVs

gg.bar.core.its.asvs <- plot_bar(all.state.asvs.fung.1920.pa, fill="Phylum")+
   geom_bar(stat="identity", position="stack")+
   scale_fill_manual(values = natparks.pals("DeathValley", 10)[2:10])+
   facet_grid(~state, scales = "free", space = "free")+
   scale_y_continuous(limits = c(0,100))+
   ggtitle("Cosmopolitan Eukaryotic ASVs")+
   theme(axis.text.x = element_blank())+theme_bw()
gg.bar.core.its.asvs
```


#### session info
```{r}
sessionInfo()
```